### ISSUE TITLE
PyLint does not recognize inherited member of a class and falsely reports no-member

### ISSUE THREAD

**Issue by adiel91** [pylint-dev/pylint#10163]
### Bug description

```python
Having 2 classes, each defined in different file, and one class inherits the other -
PyLint falsely reports no-member when analyzing only the child class file.
If giving pylint both files to analyze, the false error disappears - but only on simple cases. i have more complex files in which this is not resolved even if passing both files for analysis.

It seems pylint also reports, falsely, about "unable to import module bla" - while its not the case (running the file manually - no import issues).

In addition - if we override the member after usage, pylint will also falsely report "access-member-before-definition" -
Although the member was defined in parent class.


Here's the code used to reproduce the issue:

/tmp/adiel/bla.py:

class Test:
    def __init__(self):
        self._test = ""

/tmp/new_adiel/new_bla.py:

#!/usr/bin/env python3

import sys 
sys.path.append("/tmp/adiel")
from bla import Test

class NewTest(Test):
    def __init__(self):
        super().__init__()
        self._test += "asfasfasf"
        print(self._test)

NewTest()
```

### Configuration

```ini

```

### Command used

```shell
python3 -m pylint /tmp/new_adiel/new_bla.py
```

### Pylint output

```python
************* Module new_bla
new_adiel/new_bla.py:11:0: C0305: Trailing newlines (trailing-newlines)
new_adiel/new_bla.py:1:0: C0114: Missing module docstring (missing-module-docstring)
new_adiel/new_bla.py:5:0: E0401: Unable to import 'bla' (import-error)
new_adiel/new_bla.py:5:0: C0413: Import "from bla import Test" should be placed at the top of the module (wrong-import-position)
new_adiel/new_bla.py:7:0: C0115: Missing class docstring (missing-class-docstring)
new_adiel/new_bla.py:10:8: E1101: Instance of 'NewTest' has no '_test' member (no-member)
new_adiel/new_bla.py:7:0: R0903: Too few public methods (0/2) (too-few-public-methods)
```

### Expected behavior

Do not report no-member falsely.

### Pylint version

```shell
pylint 3.3.3
astroid 3.3.8
Python 3.11.8 (main, Dec 17 2024, 11:53:34) [GCC 12.2.0]
```

### OS / Environment

Debian 12 (Bookworm)

### Additional dependencies

```python

```

**Comment by DanielNoord** [comment]
This won't be fixed sadly. Modifying `sys.path` in the file like this is something that `pylint` will never pick up on, and is also not advised.
If the modules are importable without `sys.path` modification `pylint` will also find them and no longer gives these errors.

**Comment by adiel91** [comment]
> This won't be fixed sadly. Modifying `sys.path` in the file like this is something that `pylint` will never pick up on, and is also not advised. If the modules are importable without `sys.path` modification `pylint` will also find them and no longer gives these errors.

Thanks for clarifying - I was able to get a WA by defining in my cfg file:
[MAIN]
source-roots=<FOLDER appended to sys.path>

Since it supports relative paths. it resolved the issue.
Many thanks
